### ISSUE TITLE
Add parentheses to Classes completions

### ISSUE THREAD

**Issue by sandenbergmelo** [microsoft/pylance-release#2911]
Issue Type: Feature Request

The extension has the option __*Add parentheses to functions completions*__
It would be nice if it had this option but for classes

**Comment by debonte** [comment]
Maybe the existing setting should do this for classes, since class instantiation uses function syntax?

**Comment by sandenbergmelo** [comment]
For me only complete for functions

**Comment by erictraut** [comment]
When a partially-typed expression refers to a function, there's a very high chance that it's going to be called, so adding parentheses in this case makes a lot of sense. In contrast, classes are more likely to be used in ways that don't involve calls, so adding parentheses is more problematic in this case. It will result in extra work (needing to manually delete the inserted parentheses) a larger percentage of the time.

**Comment by debonte** [comment]
I see. And you're saying that we can't tell at completion time whether the class name is likely to be used to instantiate the class or for some other purpose? I honestly have no idea how that code works. But what about a case like this, where the class is the right-hand side of an assignment?: `x = MyClass()`

**Comment by erictraut** [comment]
Yeah, we could use some heuristics, but they would be incorrect some of the time. For example, what if you're within a call expression `x = foo(MyClass` ? Maybe we could infer whether the expected type is a class or an object in this case, but that requires type annotations on the call.

My concern is that with complex heuristics like this, the behavior will appear to be inconsistent — and perhaps even somewhat random, or at least difficult to explain.
